third_party/bigframes_vendored/sklearn/decomposition/_mf.py

```diff
@@ -0,0 +1,90 @@
+""" Matrix Factorization.
+"""
+
+# Author: Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#         Olivier Grisel <olivier.grisel@ensta.org>
+#         Mathieu Blondel <mathieu@mblondel.org>
+#         Denis A. Engemann <denis-alexander.engemann@inria.fr>
+#         Michael Eickenberg <michael.eickenberg@inria.fr>
+#         Giorgio Patrini <giorgio.patrini@anu.edu.au>
+#
+# License: BSD 3 clause
+
+from abc import ABCMeta
+
+from bigframes_vendored.sklearn.base import BaseEstimator
+
+from bigframes import constants
+
+
+class MatrixFactorization(BaseEstimator, metaclass=ABCMeta):
+    """Matrix Factorization (MF).
+
+    **Examples:**
+
+        >>> import bigframes.pandas as bpd
+        >>> from bigframes.ml.decomposition import MatrixFactorization
+        >>> X = bpd.DataFrame([[1, 1], [2, 1], [3, 1.2], [4, 1], [5, 0.8], [6, 1]])
```

Comment:
We need the cell based decomposition e.g. (row, column, value) version of a matrix to be usable in BigQuery ML and bigframes.

Let's try that. This DataFrame would be decomposed to the following:

```suggestion
        >>> X = bpd.DataFrame({
        ... "row": [0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6],
        ... "column": [0, 1] * 6,
        ... "value": [1, 1, 2, 1, 3, 1.2, 4, 1, 5, 0.8, 6, 1],
        ... })
```